project/frontend/src/containers/MySGonks/MySGonks.js

```diff
@@ -3,25 +3,106 @@ import classes from "./MySGonks.module.css";
 import Block from "../../components/UI/Block/Block";
 import LinkButton from "../../components/UI/LinkButton/LinkButton";
 import LongSGonksList from "../../components/SGonksLists/LongSGonksList/LongSGonksList";
+import ChartCard from "../../components/ChartCard/ChartCard";
 
 const MySGonks = (props) => {
+  const ONE_WEEK_MILLISECONDS = 7 * 24 * 60 * 60 * 1000;
+  const ONE_DAY_MILLISECONDS = 24 * 60 * 60 * 1000;
+  const INITIAL_WORTH = 500;
+
   const unsoldInvestments = !props.investments
     ? undefined
     : props.investments.filter(
-        (investment) => investment.dateSoldMilliSeconds == 0
+        (investment) => investment.dateSoldMilliSeconds === 0
       );
 
+  const getUsername = () => {
+    const rank = props.generalInfo.rank - 1
+    return props.rankings[rank].name
+  }
+
+  const getEarliestDate = (investments) => {
+    var currentEarliest = Infinity;
```

Comment:
Can investments be empty?

This function does slightly more than getting the earliest date,  getEarliestDateFor{Purpose} or move the ONE_WEEK_MILLISECONDS logic where it is need

Reply:
It gets the earliest date for which we have data, I'm not sure what you mean by it does more than that. I could add a docstring.?